src/olympia/addons/fields.py

```diff
@@ -291,6 +293,28 @@ def to_internal_value(self, data):
                     gettext('Unknown min app version specified')
                 )
 
+            # For creations, at this point we have a potentially incomplete
+            # ApplicationsVersions instance, which could be missing its min
+            # and/or max. In Version.from_upload() we'll fill in the gaps with
+            # the manifest data, but we can't do that here, so we only validate
+            # the range for Android if we had both a min and a max provided.
+            # ApplicationsVersions.save() will silently correct min/max before
+            # saving if it's part of the forbidden range for Android anyway.
+            if (
```

Comment:
So, if the request specifies `'min': '48.0` but no max, so the default of `'*'` is used, this exception won't raise, and it'll instead be silently corrected to X? 
What would X be?  And would `min` be respected?

Reply:
I can see the justification that if you don't specify a min or max the default isn't defined so correcting is fine (though still a bit magical).  Silently overriding the manifest min/max and having different behaviour if the developer makes a decision to explicitly define min or max in the manifest, compared to if the developer makes a decision to explicitly define min or max via the API... seems inconsistent.

Reply:
Idea: if we move the validation into the serializer - where the manifest data is available - we could calculate the resulting range based on the API, manifest, and defaults, and error if the resulting range is forbidden (perhaps generalizing `version_range_contains_forbidden_compatibility` into a utility function).

Reply:
I have been looking into this. The manifest data is available, but we would need to move the logic in `Version.from_upload()` that merges `parsed_data` with `compatibility`/`selected_apps`... That'd be quite the refactor.

Reply:
@eviljeff I don't think it's worth spending too much time/effort on this. Once Fenix allows all extensions this won't really matter (we won't actively support older Fenix versions for long).

Reply:
I'm still not a fan of the inconsistent behaviour between min/max provided in the API request JSON and the manifest.  That `'min': '48.0', 'max': '*'` in the JSON would raise, but `'min': '48.0'` in the API JSON and `'strict_max_version': '*'` in the manifest JSON wouldn't. (right?).  And just `'min': '48.0'`, leaving the default as '*', wouldn't raise either.

Would `'min' in min_max or 'max' in min_max` be more consistent, I wonder?  (it wouldn't take into account the manifest data, but it would force you to specify a matching compliant `max` and/or `min` if you provided one of them.)

Reply:
(I agree that refactoring `from_upload` is too much)

Reply:
I like the idea of doing `'min' in min_max or 'max' in min_max`, yes. Forcing you to specify a matching compliant `max` and/or `min` if you provided one is appealing. Going to look into that solution.